--- a/arch/arm64/boot/dts/rockchip/rk3399-doornet2.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-doornet2.dtsi
@@ -138,30 +138,14 @@
 	clock_in_out = "input";
 	phy-supply = <&vcc3v3_s3>;
 	phy-mode = "rgmii";
-	phy-handle = <&rtl8211f>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&rgmii_pins>;
-	snps,aal;
-	tx_delay = <0x37>;
-	rx_delay = <0x11>;
-	status = "okay";
-	
-	mdio {
-		compatible = "snps,dwmac-mdio";
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		rtl8211f: ethernet-phy@1 {
-			compatible = "ethernet-phy-id001c.c916",
-				     "ethernet-phy-ieee802.3-c22";			
-			reg = <0>;
-			pinctrl-0 = <&eth_phy_reset_pin>;
-			pinctrl-names = "default";
-			reset-assert-us = <10000>;
-			reset-deassert-us = <50000>;
-			reset-gpios = <&gpio3 RK_PB7 GPIO_ACTIVE_LOW>;
-		};
-	};
+	snps,reset-gpio = <&gpio3 RK_PB7 GPIO_ACTIVE_LOW>;
+	snps,reset-active-low;
+	snps,reset-delays-us = <0 100000 50000>;
+	tx_delay = <0x14>;
+	rx_delay = <0x0f>;
+	status = "okay";
 };
 
 &gpu {
@@ -541,10 +525,8 @@
 
 &sdhci {
 	bus-width = <8>;
-	supports-emmc;
-	assigned-clock-rates = <150000000>;
-	mmc-hs400-1_8v;
-	mmc-hs400-enhanced-strobe;
+	mmc-ddr-1_8v;
+	mmc-hs200-1_8v;
 	non-removable;
 	status = "okay";
 };

--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_mdio.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_mdio.c
@@ -363,6 +363,11 @@ int stmmac_mdio_register(struct net_device *ndev)
 		goto bus_register_fail;
 	}

+		stmmac_mdio_write(new_bus,0,31,2627);
+		stmmac_mdio_write(new_bus,0,25,0x1801);
+		stmmac_mdio_write(new_bus,0,31,0);
+		stmmac_mdio_write(new_bus,0,0,0x8000);
+
 	if (priv->plat->phy_node || mdio_node)
 		goto bus_register_done;


--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -2191,6 +2191,8 @@ static int stmmac_init_dma_engine(struct stmmac_priv *priv)
 	if (priv->extend_desc && (priv->mode == STMMAC_RING_MODE))
 		atds = 1;
 
+	msleep(1500);
+
 	ret = stmmac_reset(priv, priv->ioaddr);
 	if (ret) {
 		dev_err(priv->device, "Failed to reset the dma\n");
